src/python/pants/backend/scala/subsystems/scalac.py

```diff
@@ -35,6 +47,10 @@ class Scalac(Subsystem):
         ),
     )
 
+
+    def parsed_args_for_resolve(self, resolve_name: str) -> list[str]:
+        return self.args_for_resolve.get(resolve_name, []) + list(self.args)
```

Comment:
Should `--scala-args` come first? That way, resolve-specific args could potentially "override" the `--scala-args` (assuming scala supports that).

Reply:
Yeah, what does scalac do with repeated arguments? I don't think we need to detect or resolve this if it happens, but if it does something sensible (i.e., lets the later args override the earlier ones) then we probably want to lean in to that.